File: searchRange.cpp
class Solution {
public:
int largestArray(vector<int>&vc,int target,bool s){
    int left=0;
    int right=vc.size()-1;
    int ans=-1;
    while(left<=right){
        int mid=(left+right)/2;
        if(vc[mid]==target){
            ans=mid;
            if(!s){
                right=mid-1;
            }
            else{
                left=mid+1;
            }
        }
        else if(vc[mid]>target){
            right=mid-1;
        }
        else{
            left=mid+1;
        }
    }
    return ans;
}
    vector<int> searchRange(vector<int>& nums, int target) {
        vector<int> ans(2);
        ans[1]=(largestArray(nums,target,false));
        ans[0]=(largestArray(nums,target,true));
        return ans;
    }
};

File: searchRange.java
class Solution {
    private static int startIndex(int [] arr,int target){
        int index=-1;
        int low = 0;
        int high = arr.length-1;
        while(low<=high){
            int mid = low + (high-low)/2;
            if(arr[mid]==target){
                index=mid;
                high = mid-1;
            }
            else if(arr[mid]<target){
                low=mid+1;
            }
            else high = mid-1;
        }
        return index;

    }
    private static int endIndex(int [] arr,int target){
        int low = 0;
        int high = arr.length-1;
        int index=-1;
        while(low<=high){
            int mid = low + (high-low)/2;
            if(arr[mid]==target){
                index=mid;
                low=mid+1;
            }
            else if(arr[mid]<target) low=mid+1;
            else high=mid-1;
        }
        return index;
    }
    public int[] searchRange(int[] nums, int target) {
        int arr[]= new int[2];
        arr[0] = startIndex(nums,target);
        arr[1] = endIndex(nums,target);
        return arr;
    }
}
